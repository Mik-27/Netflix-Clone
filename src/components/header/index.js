import React, { useState } from "react";
import { Link } from "react-router-dom";
import {
	Background,
	Logo,
	Container,
	Group,
	Profile,
	Dropdown,
	Picture,
	ButtonLink,
	Feature,
	Text,
	FeatureCallout,
	TextLink,
	Search,
	SearchIcon,
	SearchInput,
	PlayButton,
} from "./styles";

export default function Header({ bg = true, children, ...restProps }) {
	return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
	return <Feature {...restProps}>{children}</Feature>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
	return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
	return <Group {...restProps}>{children}</Group>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
	return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
	return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
	return (
		<Link to={to}>
			<Logo {...restProps} />
		</Link>
	);
};

Header.ButtonLink = function HeadeerButtonLink({ children, ...restProps }) {
	return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
	return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Search = function HeaderSearch({
	searchTerm,
	setSearchTerm,
	...restProps
}) {
	const [searchActive, setSearchActive] = useState(false);
	return (
		<Search {...restProps}>
			<SearchIcon
				onClick={() => {
					// console.log(searchActive);
					setSearchActive((searchActive) => !searchActive);
				}}
			>
				<img src="/images/icons/search.png" alt="Search" />
			</SearchIcon>
			<SearchInput
				value={searchTerm}
				onChange={({ target }) => setSearchTerm(target.value)}
				placeholder="Search films and series"
				active={searchActive}
			/>
		</Search>
	);
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
	return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
	return <TextLink {...restProps}>{children}</TextLink>;
};

Header.FeatureCallout = function HeaderFeatureCallout({
	children,
	...restProps
}) {
	return <FeatureCallout {...restProps}>{children}</FeatureCallout>;
};
