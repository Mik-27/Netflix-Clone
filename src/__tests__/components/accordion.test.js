import React from "react";
import { render, fireEvent } from "@testing-library/react";
import faqsData from "../../fixtures/faqs";
import { Accordion } from "../../components";

describe("<Accordion />", () => {
	it("Renders the <Accordion /> with populated data.", () => {
		const { container, getByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{faqsData.map((item) => {
					return (
						<Accordion.Item key={item.id}>
							<Accordion.Header>{item.header}</Accordion.Header>
							<Accordion.Body>{item.body}</Accordion.Body>
						</Accordion.Item>
					);
				})}
			</Accordion>
		);

		expect(getByText("Frequently Asked Questions")).toBeTruthy();
		expect(getByText("What is Netflix?")).toBeTruthy();
		expect(getByText("How much does Netflix cost?")).toBeTruthy();
		expect(getByText("Where can I watch?")).toBeTruthy();
		expect(getByText("How do I cancel?")).toBeTruthy();
		expect(getByText("What can I watch on Netflix?")).toBeTruthy();

		expect(container.firstChild).toMatchSnapshot();
	});

	it("Opens and closes the <Accordion /> component", () => {
		const { container, queryByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{faqsData.map((item) => {
					return (
						<Accordion.Item key={item.id}>
							<Accordion.Header>{item.header}</Accordion.Header>
							<Accordion.Body>{item.body}</Accordion.Body>
						</Accordion.Item>
					);
				})}
			</Accordion>
		);

		const howMuchDoesNetflixCostText =
			"Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from Â£5.99 a month. No extra costs or contracts.";

		expect(queryByText(howMuchDoesNetflixCostText)).toBeFalsy();
		fireEvent.click(queryByText("How much does Netflix cost?"));
		expect(queryByText(howMuchDoesNetflixCostText)).toBeTruthy();

		fireEvent.click(queryByText("How much does Netflix cost?"));
		expect(queryByText(howMuchDoesNetflixCostText)).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
