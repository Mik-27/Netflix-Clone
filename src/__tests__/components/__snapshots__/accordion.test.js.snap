// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> Opens and closes the <Accordion /> component 1`] = `
<div
  class="styles__Container-innt4d-0 bwtLmJ"
>
  <div
    class="styles__Inner-innt4d-1 hsBetW"
  >
    <div
      class="styles__Title-innt4d-3 fLzTDa"
    >
      Frequently Asked Questions
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> Renders the <Accordion /> with populated data. 1`] = `
<div
  class="styles__Container-innt4d-0 bwtLmJ"
>
  <div
    class="styles__Inner-innt4d-1 hsBetW"
  >
    <div
      class="styles__Title-innt4d-3 fLzTDa"
    >
      Frequently Asked Questions
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="styles__Item-innt4d-2 iSKrdL"
    >
      <div
        class="styles__Header-innt4d-4 ccynoV"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
